File: Oficina de Programção/Nv Medio/aula_02.c
#include <stdio.h>
#include <stdlib.h>

#define MAX_SIZE 30

int array[MAX_SIZE];
int size = 0;

void menu()
{
    printf("\n============================\n");
    printf("===== 1 - Inserir array =====\n");
    printf("===== 2 - Visualizar array ==\n");
    printf("===== 3 - Pesquisar no array \n");
    printf("===== 4 - Sair ==============\n");
    printf("============================\n");
}

void insert_array()
{
    printf("\nQuantos elementos deseja inserir (máximo %d)? ", MAX_SIZE);
    scanf("%d", &size);
    
    if(size <= 0 || size > MAX_SIZE) {
        printf("Tamanho inválido!\n");
        size = 0;
        return;
    }
    
    printf("Insira %d elementos:\n", size);
    for(int i = 0; i < size; i++) {
        printf("Elemento %d: ", i+1);
        scanf("%d", &array[i]);
    }
}

void view_array()
{
    if(size == 0) {
        printf("Array vazio!\n");
        return;
    }
    
    printf("\nArray atual (%d elementos):\n", size);
    for(int i = 0; i < size; i++) {
        printf("%d ", array[i]);
    }
    printf("\n");
}

void search_array()
{
    if(size == 0) {
        printf("Array vazio! Não é possível pesquisar.\n");
        return;
    }
    
    int target;
    printf("\nDigite o valor a ser pesquisado: ");
    scanf("%d", &target);
    
    printf("Valor %d encontrado na posicao: ", target);
    int found = 0;
    for(int i = 0; i < size; i++) {
        if(array[i] == target) {
            printf("%d ", i);
            found = 1;
        }
    }
    
    if(!found) {
        printf("Nenhuma ocorrência encontrada.");
    }
    printf("\n");
}

int main()
{
    int escolha;
    
    do {
        menu();
        scanf("%d", &escolha);
        
        switch(escolha) {
            case 1:
                insert_array();
                break;
            case 2:
                view_array();
                break;
            case 3:
                search_array();
                break;
            case 4:
                printf("Saindo...\n");
                break;
            default:
                printf("Opção inválida!\n");
        }
    } while(escolha != 4);
    
    return 0;
}

File: Oficina de Programção/Nv Medio/aula_02.py
def main():
    # Função para ler o array de números inteiros
    def ler_array():
        print("Digite os números do array separados por espaço:")
        entrada = input().split()
        array = []
        for item in entrada:
            try:
                array.append(int(item))
            except ValueError:
                print(f"Valor '{item}' ignorado (não é um número inteiro válido)")
        return array

    # Função para pesquisar um item no array
    def pesquisar_array(arr):
        try:
            numero = int(input("Digite o número que deseja pesquisar: "))
            if numero in arr:
                indices = [i for i, x in enumerate(arr) if x == numero]
                print(f"O número {numero} foi encontrado no array.")
                print(f"Posições onde aparece: {indices}")
                print(f"Quantidade de ocorrências: {len(indices)}")
            else:
                print(f"O número {numero} não foi encontrado no array.")
        except ValueError:
            print("Por favor, digite um número inteiro válido.")

    # Programa principal
    print("=== Leitura e Pesquisa de Array ===")
    array_numeros = ler_array()
    
    if not array_numeros:
        print("O array está vazio. Nenhuma pesquisa pode ser realizada.")
        return
    
    print("\nArray criado:", array_numeros)
    
    while True:
        print("\nOpções:")
        print("1 - Pesquisar um número no array")
        print("2 - Mostrar o array atual")
        print("3 - Sair")
        
        opcao = input("Escolha uma opção: ")
        
        if opcao == '1':
            pesquisar_array(array_numeros)
        elif opcao == '2':
            print("Array atual:", array_numeros)
        elif opcao == '3':
            print("Programa encerrado.")
            break
        else:
            print("Opção inválida. Por favor, escolha 1, 2 ou 3.")

if __name__ == "__main__":
    main()